docs/search.rst

```diff
@@ -0,0 +1,152 @@
+:Authors:
+    Mike Cantelon
+
+Search API
+================================================================================
+
+In addition to the search functionality present in the web interface, the
+storage service also includes a REST search API. Searches are performed by
+sending an HTTP GET request.
+
+Search results will include a count of how many items were found and will
+include next and previous properties indicating links to more items in the
+result set.
+
+Location search
+--------------------------------------------------------------------------------
+
+The endpoint for searching locations is::
+
+    http://<storage service URL>/api/v2/search/location/
+
+Locations can be searched using the following search parameters:
+
+* uuid (location UUID)
+* space (space UUID)
+* purpose (purpose code)
+* enabled (whether the location is enabled)
+
+For example, if you wanted to get details about the transfer source location
+contained in the space 6d0b6cce-4372-4ef8-bf48-ce642761fd41 you could HTTP get::
```

Comment:
I noticed that if you pass a nonsense param to one of these search endpoints, you get all of the resources. Is that expected/desired? For example http://192.168.168.192:8000/api/v2/search/location/?moonbeam=figaro returns all locations. As does http://192.168.168.192:8000/api/v2/search/location/